### PR TITLE
feat(core): implement responsive popover

### PR DESCRIPTION
**Pull request by aberikashvili**
#### Please provide a link to the associated issue.
Closes SAP/fundamental-ngx #5295

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [N/A] update `README.md`
- [N/A] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/core/src/lib/popover/popover.component.ts
Comment (dimamarksman) [inline libs/core/src/lib/popover/popover.component.ts:73]:
```suggestion
    /** Whether the popover component should be displayed in mobile mode. */
```

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts
Comment (dimamarksman) [inline libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts:70]: it looks like `this.childContent` is always undefined. If not then it's not clear from where this value is getting set up.

File: libs/core/src/lib/datetime/public_api.ts
Comment (dimamarksman) [inline libs/core/src/lib/datetime/public_api.ts:8]: do we really need to keep it in the public?

Comment (aberikashvili) [reply libs/core/src/lib/datetime/public_api.ts:8]: @dimamarksman yes, me and @N1XUS were struggling to find why `npm run build-pack-library` was failing for me and this is one of the things we found

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts
Comment (aberikashvili) [reply libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts:70]: from `popover.component.ts`, where mobile component is created:
image

File: libs/core/src/lib/popover/popover.component.spec.ts
Comment (N1XUS) [inline libs/core/src/lib/popover/popover.component.spec.ts:7]: Please re-arrange imports so they follow such order
1. Imports from Angular
2. Imports from third-party libs
3. Import from local src

File: libs/core/src/lib/popover/popover.component.ts
Comment (N1XUS) [inline libs/core/src/lib/popover/popover.component.ts:34]: Re-arrange imports

File: apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.html
Comment (kavya-b) [inline apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.html:6]: why is `style="display: block">` needed?

File: apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.html
Comment (kavya-b) [inline apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.html:20]:
```suggestion
                <button fd-button (click)="addClick()" fdType="emphasized">Add Click</button>
```
same in other places

File: apps/docs/src/app/core/component-docs/popover/popover-docs.module.ts
Comment (kavya-b) [inline apps/docs/src/app/core/component-docs/popover/popover-docs.module.ts:79]: can remove duplicate `PopoverTriggerExampleComponent`

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts
Comment (kavya-b) [inline libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts:41]: can this be moved after the `public` fields and before `private` fields?

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts
Comment (kavya-b) [inline libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts:54]: `@hidden` annotation?

File: libs/core/src/lib/popover/popover.component.ts
Comment (kavya-b) [inline libs/core/src/lib/popover/popover.component.ts:114]: `@hidden`

**Review by kavya-b — COMMENTED** [review]
Looks good, few minor comments.

**Comment by InnaAtanasova** [comment]
Screen Shot 2021-06-28 at 9 58 36 AM

1. The dialog is missing the rounded corners
2. The dialog has a `grab` cursor

File: apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.html
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.html:32]: binding for fdType

File: apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.scss
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.scss:7]: You can't have this styling here. We are not allowed to target html elements like `div` and `button`. This may lead to style leaking.

Comment (InnaAtanasova) [reply apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.scss:7]: Also, this styling will not be available to the user

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.spec.ts
Comment (InnaAtanasova) [inline libs/core/src/lib/popover/popover-mobile/popover-mobile.component.spec.ts:19]: why do you have this?

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.spec.ts
Comment (InnaAtanasova) [inline libs/core/src/lib/popover/popover-mobile/popover-mobile.component.spec.ts:25]: fdType="positive"

File: libs/core/src/lib/popover/popover.component.ts
Comment (InnaAtanasova) [inline libs/core/src/lib/popover/popover.component.ts:99]: typo for `@hidden`

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts
Comment (dimamarksman) [reply libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts:70]: what magic, it's so not obvious.
in the begining we have declaration `childContent: TemplateRef<any> = undefined;` and here we are saying that it is a `TemplateRef`.
Can we drop `childContent: TemplateRef<any> = undefined;` on the line 37 since it doesn't make sense?

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts
Comment (aberikashvili) [reply libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts:54]: It's a typedoc comment

File: libs/core/src/lib/popover/popover.component.ts
Comment (aberikashvili) [reply libs/core/src/lib/popover/popover.component.ts:114]: typedoc

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.spec.ts
Comment (aberikashvili) [reply libs/core/src/lib/popover/popover-mobile/popover-mobile.component.spec.ts:19]: `menu-mobile` and `select-mobile` also use this approach, these components use service injected on component level instead of module level, to test service method calls this is the only approach I could make work

File: apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.scss
Comment (aberikashvili) [reply apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.scss:7]: @InnaAtanasova do you recommend to remove this styling or move it somewhere else?
I think once it is for temporary content, it's okay if I use `style` attribute in html

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.scss
Comment (aberikashvili) [reply libs/core/src/lib/popover/popover-mobile/popover-mobile.component.scss]: Removed, the style already exists in fundamental-styles

File: apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.scss
Comment (aberikashvili) [reply apps/docs/src/app/core/component-docs/popover/examples/popover-mobile/popover-mobile-example.component.scss]: Removed, styles

File: libs/core/src/lib/popover/popover.component.ts
Comment (aberikashvili) [reply libs/core/src/lib/popover/popover.component.ts:114]: added `@hidden` for the constructor

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts
Comment (InnaAtanasova) [reply libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts:54]: `/**@hidden */` you need a space in front of @hidden, won't work without it

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts
Comment (InnaAtanasova) [reply libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts:70]: +1 for Dima's comment

**Comment by InnaAtanasova** [comment]
@aberikashvili let 2 minor comments. In future, please do not resolve the comments yourself, usually the person who left the comments does it after they verify if they have been addressed or explained. Thanks

**Comment by aberikashvili** [comment]
> @aberikashvili let 2 minor comments. In future, please do not resolve the comments yourself, usually the person who left the comments does it after they verify if they have been addressed or explained. Thanks

okay

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts
Comment (kavya-b) [inline libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts:21]: please rearrange the imports here.

File: libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts
Comment (aberikashvili) [reply libs/core/src/lib/popover/popover-mobile/popover-mobile.component.ts]: @dimamarksman done, please check

**Review by kavya-b — APPROVED** [review]
LGTM, my comments have been addressed, thanks!

**Review by dimamarksman — APPROVED** [review]

**Comment by dimamarksman** [comment]
@InnaAtanasova waiting for your approval to merge it

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**
